content/en/blog/_posts/2021-12-15-pod-security-admission-beta.md

```diff
@@ -0,0 +1,774 @@
+```
+
+Let's figure out why the container is not starting with `kubectl -n verify-pod-security describe pod busybox-restricted`. The output is similar to this:
+
+```
+Events:
+  Type     Reason     Age                    From               Message
+  ----     ------     ----                   ----               -------
+  Warning  Failed     2m29s (x8 over 3m55s)  kubelet            Error: container has runAsNonRoot and image will run as root (pod: "busybox-restricted_verify-pod-security(a4c6a62d-2166-41a9-b288-20df17cf5c90)", container: busybox)
+```
+
+To solve this, set the effective UID (`runAsUser`) to a non-zero (root) value or use the `nobody` UID (65534).
+
+```shell
+# delete the original pod
+kubectl -n verify-pod-security delete pod busybox-restricted
+
+# create the pod again with new runAsUser
+cat <<EOF | kubectl -n verify-pod-security apply -f -
+apiVersion: v1
+kind: Pod
+metadata:
+  name: busybox-restricted
+spec:
+  securityContext:
+    runAsUser: 65534
+  containers:
+  - name: busybox
+    image: busybox
+    args:
+    - sleep
+    - "1000000"
+    securityContext:
+      seccompProfile:
+        type: RuntimeDefault
+      runAsNonRoot: true
+      allowPrivilegeEscalation: false
+      capabilities:
+        drop:
+          - ALL
+        add:
+          - NET_BIND_SERVICE
+EOF
+```
+
+Run `kubectl -n verify-pod-security get pods` to verify it is running. The output is similar to this:
+
+```
+NAME                 READY   STATUS    RESTARTS   AGE
+busybox-restricted   1/1     Running   0          25s
+```
+
+Clean up the demo (restricted pod and namespace) with:
+
+```shell
+kubectl delete namespace verify-pod-security
+```
+
+At this point, if you wanted to dive deeper into linux permissions or what is permitted for a certain container, exec into the control plane and play around with `containerd` and `crictl inspect`.
+
+```shell
+# if using docker, shell into the control plane
+docker exec -it kind-control-plane bash
+
+# list running containers
+crictl ps
+
+# inspect each one by container ID
+crictl inspect <CONTAINER ID>
+```
+
+### Applying a cluster-wide policy
+
+In addition to applying labels to namespaces to configure policy you can also configure cluster-wide policies and exemptions using the AdmissionConfiguration resource. 
+
+Using this resource, policy definitions are applied cluster-wide by default and any policy that is applied via namespace labels will take precedence. 
+
+There is no runtime configurable API for this resource and a cluster administrator would need to specify a path to the file below via the `--admission-control-config-file` flag on the API server. 
+
+In the following resource we are enforcing the baseline policy and warning and auditing the baseline policy. We are also making the kube-system namespace exempt from this policy.
+
+It's not recommended to alter control plane / clusters after install, so let's build a new cluster with a default policy on all namespaces.
```

Comment:
on the other hand, this might be a chance to show up the safe rollout capabilities of this feature... you can tighten default warn/audit levels on an existing cluster without breaking writes to existing pods or workloads.